src/jit/compiler.hpp

```diff
@@ -500,6 +500,52 @@ inline regNumber genRegNumFromMask(regMaskTP mask)
     return regNum;
 }
 
+//------------------------------------------------------------------------------
+// genTypeValueFitsIn : Checks if a value can be represented by a given type.
```

Comment:
I think I'll go with `genTypeCanRepresentValue` in an attempt to maintain consistency, we already have a bunch of `genTypeX` functions. Then I can also pass the type as the first argument, seems better that way.